sdk/src/main/java/com/hedera/hashgraph/sdk/Transaction.java

```diff
@@ -1496,4 +1488,211 @@ public int getTransactionBodySize() {
+
+        public AccountId getNodeID() {
+            return nodeID;
+        }
+
+        public TransactionId getTransactionID() {
+            return transactionID;
+        }
+
+        public byte[] getBody() {
+            return body;
+        }
+    }
+
+    /**
+     * Returns a list of SignableNodeTransactionBodyBytes objects for each signed transaction in the transaction list.
+     * The NodeID represents the node that this transaction is signed for.
+     * The TransactionID is useful for signing chunked transactions like FileAppendTransaction,
+     * since they can have multiple transaction ids.
+     *
+     * @return List of SignableNodeTransactionBodyBytes
+     * @throws RuntimeException if transaction is not frozen or protobuf parsing fails
+     */
+    public List<SignableNodeTransactionBodyBytes> getSignableNodeBodyBytesList() {
+        if (!this.isFrozen()) {
+            throw new RuntimeException("Transaction is not frozen");
+        }
+
+        List<SignableNodeTransactionBodyBytes> signableNodeTransactionBodyBytesList = new ArrayList<>();
+
+        for (int i = 0; i < innerSignedTransactions.size(); i++) {
+            SignedTransaction signableNodeTransactionBodyBytes =
+                    innerSignedTransactions.get(i).build();
+
+            TransactionBody body;
+
+            body = parseTransactionBody(signableNodeTransactionBodyBytes.getBodyBytes());
+
+            AccountId nodeID = AccountId.fromProtobuf(body.getNodeAccountID());
+            TransactionId transactionID = TransactionId.fromProtobuf(body.getTransactionID());
+
+            signableNodeTransactionBodyBytesList.add(new SignableNodeTransactionBodyBytes(
+                    nodeID,
+                    transactionID,
+                    signableNodeTransactionBodyBytes.getBodyBytes().toByteArray()));
+        }
+
+        return signableNodeTransactionBodyBytesList;
+    }
+
+    /**
+     * Adds a signature to the transaction for a specific transaction id and node id.
+     * This is useful for signing chunked transactions like FileAppendTransaction,
+     * since they can have multiple transaction ids.
+     *
+     * @param publicKey The public key to add signature for
+     * @param signature The signature bytes
+     * @param transactionID The specific transaction ID to match
+     * @param nodeID The specific node ID to match
+     * @return The child transaction (this)
+     * @throws RuntimeException if unmarshaling fails or invalid signed transaction
+     */
+    public T addSignatureV2(PublicKey publicKey, byte[] signature, TransactionId transactionID, AccountId nodeID) {
+
+        if (innerSignedTransactions.isEmpty()) {
+            // noinspection unchecked
+            return (T) this;
+        }
+
+        transactionIds.setLocked(true);
+
+        for (int index = 0; index < innerSignedTransactions.size(); index++) {
+            if (processSignatureForTransaction(index, publicKey, signature, transactionID, nodeID)) {
```

Comment:
to better indicate it's a boolean func

```suggestion
            if (processedSignatureForTransaction(index, publicKey, signature, transactionID, nodeID)) {
```